### ISSUE TITLE
[speech-to-text] Recognize method with IAM Key

### ISSUE THREAD

**Issue by chughts** [watson-developer-cloud/node-sdk#696]
- Node version : 8.11.1
- SDK version : 3.4.2

There is a conflict between the IAM Key processing and the Recognize method. The Recognize function works when the service credentials are a userid / password, but when a iam_key is required there is an exception in trying to pipe the audio stream - 

```
Error Reported  TypeError: Cannot read property 'on' of undefined
    at ReadStream.Readable.pipe (_stream_readable.js:582:8)
    at SpeechToTextV1.recognize (/Users/soheelchughtai/Documents/dev/node-red/wdc/node-red-node-watson/node_modules/watson-developer-cloud/speech-to-text/v1.js:457:14)
```

If looks as though the fetch of an IAM token interferes with the stream, but I have not been able to determine how. 

All the other methods fetch models, list customisations, create customisations etc. are working with an IAM key except recognize.

**Comment by pekevski** [comment]
@chughts What object do you pass into the constructor for SpeechToText object? I can't seem to call `createRecognizeStream()` without getting the following error:

```
speech-to-text/speech-to-text-demo/node_modules/watson-developer-cloud/speech-to-text/v1.js:397
            authorization: this._options.headers.Authorization
                                                 ^

TypeError: Cannot read property 'Authorization' of undefined
```

**Comment by chughts** [comment]
@pekevski Your error relates to using web sockets from the browser. When using 'watson-developer-cloud/authorization/v1') to fetch a token. I get the same problem, I haven't worked though that one yet, but I think that it will need to be a separate issue on this repo. 

For now I am trying to use plain old `recognize` from server-side node.js, and failing.

**Comment by germanattanasio** [comment]
@anweshan ☝️

**Comment by germanattanasio** [comment]
This should be fixed in the last release.

**Comment by germanattanasio** [comment]
fixed in `3.5.0`

**Comment by germanattanasio** [comment]
This will be fixed once https://github.com/watson-developer-cloud/node-sdk/pull/717 is merged

**Comment by dpopp07** [comment]
Resolved with #717
